oqsprov/oqsprov_keys.c

```diff
@@ -1377,6 +1377,19 @@ OQSX_KEY *oqsx_key_from_pkcs8(const PKCS8_PRIV_KEY_INFO *p8inf,
                         key_diff = nids_sig[6].length_private_key - buflen;
                 }
 
+                // removing extra OTECT STRING from ED25519 and ED448 keys
+                if ((keytype == EVP_PKEY_ED25519) ||
+                    (keytype == EVP_PKEY_ED448)) {
+                    ASN1_OCTET_STRING *ed_octet = NULL;
+                    ed_octet = d2i_ASN1_OCTET_STRING(&ed_octet, &buf, buflen);
+                    aux += ed_octet->length;
+                    memcpy(concat_key + plen - 1 - aux, ed_octet->data,
+                           ed_octet->length);
+                    nid = 1; // setting to non zero value so the key is not
```

Comment:
This seems to be a side-effect-creating approach to obtaining some functionality (or is it documented somewhere what setting `nid` effects?) and as such "brittle", don't you think @feventura ? But when trying to understand it I realized the whole PKCS#8 logic also for non-composite keys has been changed upside down with #317 already, so I guess we have to live with it for now.

Reply:
I started using `nid` ad a nid value (you can see that in 1288) but this became a check for the `memcpy` from `buf` to `concat_key` (see line 1393), right now this value behaves more as this check than an actual nid value.
Also, this value is contained inside the composite code, it is defined in line 1249, which is inside the `else` for the `if (get_keytype(OBJ_obj2nid(palg->algorithm)) != KEY_TYPE_CMP_SIG)`.
I agree that the name of this variable don't fully represent it anymore, but the effects are contained inside the composite code and represent a check that the buf value was set up.
The non-composite keys code did not change at all, I kept is as it was and its contained between lines 1228 and 1236.